libmamba/tests/src/core/test_filesystem.cpp

```diff
@@ -349,6 +351,50 @@ namespace mamba
             fs::remove_all(tmp_dir);
             REQUIRE_FALSE(fs::exists(tmp_dir));
         }
+
+        TEST_CASE("create_cache_dir")
+        {
+            const auto cache_dir = fs::temp_directory_path() / "mamba-fs-cache-dir";
+            const auto set_gid_tmp_dir = fs::temp_directory_path() / "mamba-fs-set_gid_tmp_dir";
+
+            mamba::on_scope_exit _(
+                [&]
+                {
+                    fs::remove_all(cache_dir);
+                    fs::remove_all(set_gid_tmp_dir);
+                }
+            );
+
+            // Get the information about whether the filesystem supports the `set_gid` bit.
+            bool supports_setgid_bit = false;
+
+            std::error_code ec;
+            fs::permissions(set_gid_tmp_dir, fs::perms::set_gid, fs::perm_options::add, ec);
+
+            if (!ec)
+            {
+                supports_setgid_bit = (fs::status(set_gid_tmp_dir).permissions() & fs::perms::set_gid)
+                                      == fs::perms::set_gid;
+            }
+
+            create_cache_dir(cache_dir);
+
```

Comment:
```suggestion
            REQUIRE(fs::is_directory(cache_dir));
```
It is not clear to me if `fs::status(cache_dir).permissions()` will end up in error if the directory doesnt exist after calling this functions, so in doubt I would explicitly check the assumption that if it didnt fail the directory indeed exists.
Even better if you also check that it does not exist before calling `create_cache_dir`.

This is all an improvement suggestion.